### DISCUSSION TITLE
How can I get the DataGridView Cell Value into a TextBox?

### DISCUSSION THREAD

**Discussion by MastersoftDev** [pascalabcnet/pascalabcnet#2717] [Q&A] [answered]
One more last point, can help me know how I can get the DataGridView Cell Value into a TextBox? It seems hard for me to do it. In Visual Basic.net I do it like this;

If e.RowIndex >= 0 Then
	Dim row As DataGridViewRow = purchaseItemsGridView.Rows(e.RowIndex)
	If row.Index = purchaseItemsGridView.Rows.Count - 1 Then
		Return
	End If
	txtItemNo.Text = (row.Cells(0).Value.ToString)
	txtItemName.Text = (row.Cells(2).Value.ToString)
	txtQuantity.Text = (row.Cells(3).Value.ToString)
	txtUnitCost.Text = (row.Cells(4).Value.ToString)
	txtAmount.Text = (row.Cells(5).Value.ToString)
End If
I'm failing to write the same code in PascalABCNET! Please, can you help me?

Thank you,

Adrian

**Comment by samuraiGH** [comment, marked as answer]
```pas
if e.RowIndex >= 0 then begin
  var row:= purchaseItemsGridView.Rows[e.RowIndex];
  if row.Index = purchaseItemsGridView.Rows.Count - 1 then
    exit;
  txtItemNo.Text:= row.Cells[0].Value.ToString;
  ...
end;
```

**Reply by MastersoftDev** [reply]
Thank you very much!
God Bless You in Jesus' Name.
